test/Dialect/FIRRTL/parse-basic.fir

```diff
@@ -1934,3 +1934,121 @@ circuit LayerEnabledModule:
   module UserOfLayerEnabledModule enablelayer A enablelayer B.C:
     ; CHECK: firrtl.instance i interesting_name {layers = [@A, @B::@C]} @LayerEnabledModule()
     inst i of LayerEnabledModule
+
```

Comment:
Below are the tests I added. I wrote them when doing the work. You may notice that many test the same code paths--my thinking was more defensive for future changes that perhaps some of these codepaths could end up different. In any case, I'm not wedded to them so I'm happy to pare them down if reviewers think that's better. The tests I had to update do seem to capture the 4.0.0 behavior, but obviously we do need the check for the < 4.0.0 behavior.